controller/handler_ctrl/bind.go

```diff
@@ -56,9 +64,21 @@ func (self *bindHandler) BindChannel(binding channel.Binding) error {
 	binding.AddPeekHandler(trace.NewChannelPeekHandler(self.network.GetAppId(), binding.GetChannel(), self.network.GetTraceController(), traceDispatchWrapper))
 	binding.AddPeekHandler(metrics2.NewCtrlChannelPeekHandler(self.router.Id, self.network.GetMetricsRegistry()))
 
-	if self.router.VersionInfo.HasMinimumVersion("0.18.7") {
-		roundTripHistogram := self.network.GetMetricsRegistry().Histogram("ctrl.latency:" + self.router.Id)
-		queueTimeHistogram := self.network.GetMetricsRegistry().Histogram("ctrl.queue_time:" + self.router.Id)
+	doHeartbeat := self.router.VersionInfo.HasMinimumVersion("0.25.5")
+
+	roundTripHistogram := self.network.GetMetricsRegistry().Histogram("ctrl.latency:" + self.router.Id)
+	queueTimeHistogram := self.network.GetMetricsRegistry().Histogram("ctrl.queue_time:" + self.router.Id)
+	if doHeartbeat {
+		log.Info("router supports heartbeats")
+		cb := &heartbeatCallback{
+			latencyMetric:    roundTripHistogram,
+			queueTimeMetric:  queueTimeHistogram,
+			ch:               binding.GetChannel(),
+			latencySemaphore: concurrenz.NewSemaphore(2),
+		}
+		channel.ConfigureHeartbeat(binding, 10*time.Second, time.Second, cb)
```

Comment:
We need to have the same close handler to release the metrics as we do in the else block below. I need to check the link code if that's missing there as well. It likely is

Reply:
Yes, it's missing in the handler_link/bind.go as well. Would you mind adding it there as well, please?

Reply:
Moved the close handler to be outside of the if/else block on the control channel part. Double checked the link side I think it's already done.

Reply:
:+1: Missed it in the link code, as I was looking at the bottom.